osmnx/io.py

```diff
@@ -440,18 +457,28 @@ def _convert_edge_attr_types(G, dtypes=None):
     Returns
     -------
     G : networkx.MultiDiGraph
+    
+    Modified to include detection of list, dict, and set type
+    node attribute strings and handle their translation into
+    python objects of the correct type.
     """
     # for each edge in the graph, eval attribute value lists and convert types
     for _, _, data in G.edges(data=True, keys=False):
         # remove extraneous "id" attribute added by graphml saving
         data.pop("id", None)
 
-        # first, eval stringified lists to convert them to list objects
+        # first, eval stringified lists, dicts or sets to convert them to objects
         # edge attributes might have a single value, or a list if simplified
         for attr, value in data.items():
+            # check for stringified lists
             if value.startswith("[") and value.endswith("]"):
```

Comment:
This could probably be simplified to a single `if` block like `if (value.startswith("[") and value.endswith("]")) or (value.startswith("{") and value.endswith("}")):`

Reply:
Very true.  I'd done them separately, more for readability than anything, but that's subjective at best.